### ISSUE TITLE
Fault reporting seems over eager

### ISSUE THREAD

**Issue by CaptainPants** [CaptainPants/sweeter#25]
At the moment a lot of errors thrown in component life cycle report a fault to the fault context immediately. Arguably this should not happen if the component isn't actually added to the tree. We could return a place holder / if appropriate check if the component is mounted/wait until it is.

**Comment by CaptainPants** [comment]
I would treat this as dependent on #26

**Comment by CaptainPants** [comment]
I think this is actually not true, as the fault boundaries are actually at the component level. ErrorBoundary only differs in that it catches any errors coming out of its children (as a signal) and falls back to alternate context. Closing as "Won't do".
